### PR TITLE
[inductor] use smaller RBLOCK for expensive reduction kernels

### PR DESCRIPTION
**Pull request by shunting314**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #126348
* __->__ #126477

Triton sometimes uses less registers for more expensive kernel which results in worse perf ( https://github.com/pytorch/pytorch/issues/126463 ). This may make inductor end up with a sub-optimal config. Use a smaller max RBLOCK if the reduction potentially need many registers.

Will run perf test..

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

**Review by eellison — COMMENTED** [review]
Maybe this motivation to do live variable analysis.. going to wait to see how perf results look.

File: torch/_inductor/runtime/triton_heuristics.py [hunk 1]
```diff
@@ -1480,12 +1485,32 @@ def _reduction_configs(
     assert len(size_hints) == 2
     rnumel = size_hints[-1]
 
+    MAX_RBLOCK = 2048
+    if (
```
[the remark this answers is not in the document]
Comment (shunting314) [reply torch/_inductor/runtime/triton_heuristics.py:1489]: no. I checked `CompiledKernel.n_spills`, it's 0.

**Review by jansel — APPROVED** [review]

**Comment by shunting314** [comment]
Perf test on this PR is [roughly neutral](https://hud.pytorch.org/benchmark/compilers?startTime=Wed%2C%2015%20May%202024%2018%3A49%3A51%20GMT&stopTime=Wed%2C%2022%20May%202024%2018%3A49%3A51%20GMT&granularity=hour&suite=torchbench&mode=training&dtype=amp&lBranch=gh/shunting314/146/head&lCommit=32e418d539d0a70f95ef4acb2abec0013fc160da&rBranch=main&rCommit=53f73cdeb6fe53155f91064ca15b722e80dec2f3). 

I'll land this one and debug the perf/compilation time regress for the PR tuning partitioner.

**Pull request closed without merging**